### PR TITLE
Chupiak hw

### PR DESCRIPTION
**Pull request by Chupiak**

### CODE REVIEW

File: src/makeInfinityAdder.js [hunk 1]
```diff
@@ -4,9 +4,25 @@
  *
  * @return {function}
  */
-
 function makeInfinityAdder() {
-  // write code here
+  let total = 0;
+
+  // eslint-disable-next-line no-shadow
```
Comment (DarkMistyRoom) [inline src/makeInfinityAdder.js:10]: why do you need it? make it work without

File: src/makeInfinityAdder.js [hunk 2]
```diff
@@ -4,9 +4,25 @@
...
+  function adder(item) {
+    if (item !== undefined) {
+      total = total + item;
+    } else {
+      const result = total;
+
+      total = 0;
+
+      return result;
+    }
```
Comment (DarkMistyRoom) [inline src/makeInfinityAdder.js:20]: refactor to avoid `else`. you can do the same with only one plane `if`

File: src/makeInfinityAdder.js [hunk 3]
```diff
@@ -4,9 +4,25 @@
...
  */
-
 function makeInfinityAdder() {
-  // write code here
+  let total = 0;
+
+  // eslint-disable-next-line no-shadow
+  function adder(item) {
+    if (item !== undefined) {
+      total = total + item;
```
Comment (DarkMistyRoom) [inline src/makeInfinityAdder.js:13]:
```suggestion
      total += item;
```

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]

**Comment by SanyaBratashchuk** [comment]
Why did you close this PR?

**Pull request closed without merging**